### PR TITLE
API: Endpoint to get an organization's users

### PR DESCRIPTION
**Pull request by twinsfan421**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This would provide an api endpoint `dev.to/api/organizations/{org_username}/users` to get a list of an organization's users.
As discussed in issue 9212, This is the 2nd of 4 related endpoints that will include:
```
dev.to/api/organizations/{org_username}/articles
dev.to/api/organizations/{org_username}/listings
```

## Related Tickets & Documents
#9212  #10931

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [x] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [ ] No documentation needed

### CODE REVIEW

**Review by mstruve — APPROVED** [review]
Code looks good to me. @rhymes is probably a better person to ensure the docs are good to go. Thanks for taking the time to breakout these API endpoints into their own PRs, it makes our jobs as reviewers a lot easier. We really appreciate it! 

🚀

**Comment by benhalpern** [comment]
This looks good to me in terms of the base premise. (AKA it's all public info and matches good patterns, so totally should exist)..

Also defer to @rhymes for final say on API reviews.

File: config/routes.rb [hunk 1]
```diff
@@ -204,6 +204,7 @@
         resources :organizations, only: [] do
           collection do
             get "/:org_username", to: "organizations#show"
+            get "/:org_username/users", to: "organizations#users"
           end
         end
```
Comment (rhymes) [inline config/routes.rb:209]: do you think we could use resource routes here?

For example:

```ruby
        resources :organizations, only: [:show], param: :username do
          resources :users, only: [:index], to: "organizations#users"
        end
```

(renamed the parameter to `:username` as it's implicit it's part of an org)

this will also have the effect of generating helpers such as: `api_organization_users_path` and `api_organization_path` that you can use in the tests.

Also, if the `API::OrganizationsController` becomes too big we can easily move the actions to their own resource controller (for example: `API::OrganizationsUsersController` without changing the external URLs)

File: app/controllers/api/v0/organizations_controller.rb [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
     class OrganizationsController < ApiController
+      before_action :find_organization, only: %i[users]
+
       SHOW_ATTRIBUTES_FOR_SERIALIZATION = %i[
         username name summary twitter_username github_username url
         location created_at profile_image tech_stack tag_line story
       ].freeze
       private_constant :SHOW_ATTRIBUTES_FOR_SERIALIZATION
 
+      ORG_USERS_FOR_SERIALIZATION = %i[
```
Comment (rhymes) [inline app/controllers/api/v0/organizations_controller.rb:12]: I wouldn't make a difference between "regular users" and "organization users". As I don't think we plan to tell the API what's their role (admins, non admins) for privacy reasons, I think we can render the same partial we render for `/api/users/:id` in the `Api::V0::UsersController#show` route in a loop in the jbuilder file

Let me know if that makes sense

File: app/views/api/v0/organizations/users.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,7 @@
+json.array! @users do |user|
+  json.type_of "org_user"
+  json.extract!(user, :name, :username, :twitter_username, :github_username)
+
+  json.website_url user.processed_website_url
+  json.profile_image Images::Profile.call(user.profile_image_url, length: 320)
```
Comment (rhymes) [inline app/views/api/v0/organizations/users.json.jbuilder:6]: see my point about rendering `api/v0/views/show.json.jbuilder` here (you can use a common partial for the two)

File: docs/api_v0.yml [hunk 4]
```diff
@@ -1147,6 +1147,30 @@ components:
           description: Profile image (640x640)
           format: url
 
+    OrgUser:
```
Comment (rhymes) [inline docs/api_v0.yml:1150]: if we recycle the `User` entity, this is not needed

File: docs/api_v0.yml [hunk 5]
```diff
@@ -1726,6 +1750,16 @@ components:
         story:
         profile_image: https://res.cloudinary.com/...jpeg
 
+    OrgUser:
```
Comment (rhymes) [inline docs/api_v0.yml:1753]: same as above

File: docs/api_v0.yml [hunk 6]
```diff
@@ -3023,6 +3057,54 @@ paths:
           source: |
             curl https://dev.to/api/organizations/ecorp
 
+  /organizations/{org_username}/users:
+    get:
+      operationId: getOrgUsers
+      summary: Organization's users
+      description: |
+        This endpoint allows the client to retrieve a list of an organization's users
```
Comment (rhymes) [inline docs/api_v0.yml:3065]: I would rephrase this as "users belonging to the organization"

**Review by rhymes — CHANGES_REQUESTED** [review]
Hi @twinsfan421, thanks a lot for the PR! I think we need to address a few implementation details but overall it works!

File: config/routes.rb (hunk 1 above)
Comment (twinsfan421) [reply config/routes.rb]: Ya for sure. I just pushed up an update for the routes

File: app/controllers/api/v0/organizations_controller.rb (hunk 2 above)
Comment (twinsfan421) [reply app/controllers/api/v0/organizations_controller.rb]: Makes sense. I just pushed up a fix for this

File: app/views/api/v0/organizations/users.json.jbuilder (hunk 3 above)
Comment (twinsfan421) [reply app/views/api/v0/organizations/users.json.jbuilder]: I created a shared partial and pushed it up

**Comment by twinsfan421** [comment]
Thanks @rhymes for reviewing. Let me know if you want me to change anything else

**Review by rhymes — APPROVED** [review]
@twinsfan421 it works great! 

Thank you!
